modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RestRankEvalAction.java

```diff
@@ -0,0 +1,195 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.rankeval;
+
+import org.elasticsearch.client.Client;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.rest.BaseRestHandler;
+import org.elasticsearch.rest.RestChannel;
+import org.elasticsearch.rest.RestRequest;
+
+/**
+ * Accepted input format:
+ * 
+ * General Format:
+ * 
+ * 
+  {
+    "user_requests": [{
+        "id": "human_readable_id",
+        "request_params": { ... map of template params ... },
+        "doc_ratings": { ... mapping from doc id to rating value ... }
+     }],
+
+    "translation_spec": {
+        "id": "human_readable_id",
+        "request_template": {... query template to evaluate ...}
+    },
+    "evaluation_metric": {
+        "... metric_name... ": {
+            "... metric_parameter_key ...": ...metric_parameter_value...
+        }
+    }
+  }
+
+ * 
+ * Example: 
+ * 
+ * 
+   {
+    "user_requests": [{
+        "id": "amsterdam_query",
+        "request_params": {
```

Comment:
I really wonder if we should just expect a full blown query in there instead of templates. It would simplify a lot and users can still put the template in the clusterstate to run stuff against if they use a template query. then they just put pass the params with the query as a template query and we can be our own plugin / module without the dependency